src/util/config.cpp

```diff
@@ -233,14 +234,15 @@ void configt::ansi_ct::set_arch_spec_power(const irep_idt &subarch)
   switch(mode)
   {
   case flavourt::GCC:
-  case flavourt::APPLE:
+  case flavourt::CLANG:
     defines.push_back("__powerpc");
     defines.push_back("__powerpc__");
     defines.push_back("__POWERPC__");
     defines.push_back("__ppc__");
 
-    if(mode==flavourt::APPLE)
-      defines.push_back("__BIG_ENDIAN__");
+#if defined(__APPLE__)
+    defines.push_back("__BIG_ENDIAN__");
+#endif
```

Comment:
This looks wrong to me: it introduces a compile-time configuration into what ought to be a runtime configuration choice. This needs to test for the (target) operating system (`os == "macos"`).

Reply:
Ok, fixed.